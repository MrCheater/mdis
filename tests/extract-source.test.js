const path = require('path');

const { extractSource } = require('../src/extract-source');

describe('extractSource', () => {
  test('"named-blocks.js - options = { name: undefined }"', () => {
    expect(
      extractSource(path.join(__dirname, 'files', 'named-blocks.js'))
    ).toMatchSnapshot();
  });

  test('"named-blocks.js - options = { name: "imports" }"', () => {
    expect(
      extractSource(path.join(__dirname, 'files', 'named-blocks.js'), {
        name: 'imports'
      })
    ).toMatchSnapshot();
  });

  test('"named-blocks.js - options = { name: "Sample" }"', () => {
    expect(
      extractSource(path.join(__dirname, 'files', 'named-blocks.js'), {
        name: 'Sample'
      })
    ).toMatchSnapshot();
  });

  test('"named-blocks.js - options = { name: "render" }"', () => {
    expect(
      extractSource(path.join(__dirname, 'files', 'named-blocks.js'), {
        name: 'render'
      })
    ).toMatchSnapshot();
  });

  test('"simple.js"', () => {
    expect(
      extractSource(path.join(__dirname, 'files', 'simple.js'))
    ).toMatchSnapshot();
  });

  test('"strings-and-comments.js"', () => {
    expect(
      extractSource(path.join(__dirname, 'files', 'strings-and-comments.js'))
    ).toMatchSnapshot();
  });
});
