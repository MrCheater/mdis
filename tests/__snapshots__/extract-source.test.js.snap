// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractSource "named-blocks.js - options = { name: "Sample" }" 1`] = `
"class NamedBlocks extends React.PureComponent {
  }
"
`;

exports[`extractSource "named-blocks.js - options = { name: "imports" }" 1`] = `
"import React from 'react';
"
`;

exports[`extractSource "named-blocks.js - options = { name: "render" }" 1`] = `
"  render() {
    return <div>Sample</div>;
  }
  "
`;

exports[`extractSource "named-blocks.js - options = { name: undefined }" 1`] = `""`;

exports[`extractSource "simple.js" 1`] = `
"class Point {
}
"
`;

exports[`extractSource "strings-and-comments.js" 1`] = `""`;
